src/scene/load/x3dloadinternaltiledmap.pas

```diff
@@ -48,6 +48,39 @@ function LoadTiledMap2d(const Stream: TStream; const BaseUrl: String): TX3DRootN
   { Convert Tiled map into X3D node. }
   TCastleTiledMapConverter = class
   strict private
+    type
+    TTilesetNodes = record
+      Coord: TCoordinateNode;
+      TexCoord: TTextureCoordinateNode;
+    end;
+
+    TAnimationNodes= record
+      CoordNodes: TTilesetNodes;
+      TexCoordInterp: TCoordinateInterpolator2DNode;
+    end;
+
+    TAnimationWithFlips = record
+      Animation : TCastleTiledMapData.TAnimation;
+      HorizontalFlip, VerticalFlip, DiagonalFlip: Boolean;
+    end;
+
+    { CycleInterval in milliseconds. }
+    TTimeSensorContext = {$ifdef FPC}specialize{$endif} TDictionary<Cardinal,TTimeSensorNode>;
+
+    TAnimationContext = {$ifdef FPC}specialize{$endif} TDictionary<TAnimationWithFlips,TAnimationNodes>;
+
+    TRenderContext =class ({$ifdef FPC}specialize{$endif} TDictionary<TCastleTiledMapData.TTileset,TTilesetNodes>)
```

Comment:
We already have a class `TRenderContext` in CGE (in `CastleRenderContext`), so this one should be renamed, to avoid confusion :) 

In general, the word "context" in CGE is often associated with "OpenGL(ES) rendering context", it's better not to use it for something else.

I would propose: `TLayerConversion`. As the purpose of it is to encapsulate information local to `TCastleTiledMapConverter.BuildTileLayerNode` call.

So

- TRenderContext -> TLayerConversion
- TTimeSensorContext -> TLayerTimeSensors
- TAnimationContext -> TLayerAnimations

Reply:
I dropped the idea because that would make the code too complicated.